File: Java/FactoryPattern/DatabaseFactory/src/com/laxmena/factory/NoSQLFactory.java
package com.laxmena.factory;

import com.laxmena.db.Database;
import com.laxmena.db.MongoDatabase;

public class NoSQLFactory extends DatabaseFactory {

    @Override
    public Database getDatabase(String database) {
        if(database.equalsIgnoreCase("MongoDB")) {
            return new MongoDatabase();
        } else {
            System.out.println("Database " + database + " - Not supported");
            return null;
        }
    }
}

File: Factory/Python/DatabaseAdapter/DatabaseAdapter/factory/nosql_factory.py
from db.database import Database
from db.mongo_database import MongoDatabase
from factory.database_factory import DatabaseFactory


class NoSQLFactory(DatabaseFactory):
    @staticmethod
    def create_database(name: str) -> Database:
        if name == "mongodb":
            return MongoDatabase()
        else:
            print("Requested database not found")
            return None
